### PR TITLE
Sycl queue docstrings

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Docstrings change:

   - Examples for `SyclContext` were modified to run.
   - Docstring for `SyclQueue` was modified following the lead of `SyclContext` from #385

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
Chunk of rendered doc page:

image

File: dpctl/_sycl_context.pyx [hunk 1]
```diff
@@ -125,20 +125,22 @@ cdef class SyclContext(_SyclContext):
...
                 assert(len(ctx.get_devices) == len(sub_devices))
 
-        - Invoking the constuctor with a named ``PyCapsule`` with the name
-          **SyclContextRef** that carries a pointer to a ``sycl::context``
-          object.
+        - Invoking the constuctor with a named ``PyCapsule`` with name
+          **"SyclContextRef"** that carries a pointer to a ``sycl::context``
+          object. The capsule will be renamed upon successful consumption
+          to ensure one-time use. A new named capsule can be constructed by
+          using `ctx._get_capsule()` method.
```
Comment (diptorupd) [inline dpctl/_sycl_context.pyx:137]: How about :func:`dpctl.SyclContext._get_capsule()`? We will get a link to the doc for `_get_capsule` instead of italicizing the function name.

File: dpctl/_sycl_queue.pyx [hunk 2]
```diff
@@ -158,17 +158,97 @@ cdef class _SyclQueue:
...
+                # Create a context common to all the sub-devices.
+                ctx = dpctl.SyclContext(sub_devices)
+                # create a queue for each sub-device using the common context
+                queues = [dpctl.SyclQueue(ctx, sub_d) for sub_d in sub_devices]
+
+        - Invoking the constuctor with a named ``PyCapsule`` with the name
+          **"SyclQueueRef"** that carries a pointer to a ``sycl::queue``
+          object. The capsule will be renamed upon successful consumption
+          to ensure one-time use. A new named capsule can be constructed by
+          using `q._get_capsule()` method.
```
Comment (diptorupd) [inline dpctl/_sycl_queue.pyx:226]: Same here.

File: dpctl/_sycl_queue.pyx [hunk 3]
```diff
@@ -239,16 +319,14 @@ cdef class SyclQueue(_SyclQueue):
                 raise SyclQueueCreationError(
                     "SYCL Context could not be created from '{}'.".format(arg)
                 )
-            elif status == -4:
+            elif status == -4 or status == -6:
```
Comment (diptorupd) [inline dpctl/_sycl_queue.pyx:322]: This is a non-documentation related change. Better to move to a separate commit.

Comment (oleksandr-pavlyk) [reply dpctl/_sycl_queue.pyx]: Done, moved to #389

File: dpctl/_sycl_queue.pyx (hunk 2 above)
Comment (oleksandr-pavlyk) [reply dpctl/_sycl_queue.pyx]: Done

File: dpctl/_sycl_context.pyx (hunk 1 above)
Comment (oleksandr-pavlyk) [reply dpctl/_sycl_context.pyx]: done

**Review by diptorupd — APPROVED** [review]
